interface/views.py

```diff
@@ -110,15 +109,14 @@ def download(request, pk):
             not in submission.assignment.course.teaching_assistants.all()):
         return HttpResponse(status=403)
 
-    with TemporaryDirectory() as _tmp:
-        tmp = Path(_tmp)
+    buff = io.BytesIO()
+    submission.download(buff)
+    buff.name = f'{submission.pk}.zip'
+    buff.seek(0)
 
-        submission.download(tmp / f'{submission.pk}.zip')
-        review_zip = (tmp / f'{submission.pk}.zip').open('rb')
+    log_action("Download submission", request.user, submission)
 
-        log_action("Download submission", request.user, submission)
-
-        return FileResponse(review_zip, as_attachment=True)
+    return FileResponse(buff, as_attachment=True)
```

Comment:
Let's have a common way of giving names to downloaded files. Either set filename param of `FileResponse` to `f'{submission.pk}.zip'` or set the name in buff.name, in both `interface/admin.py::zip_submissions` and here.